File: app/src/main/java/io/github/ovso/righttoknow/framework/adapter/BaseAdapterDataModel.java
package io.github.ovso.righttoknow.framework.adapter;

import java.util.List;

public interface BaseAdapterDataModel<T> {
  void add(T item);
  void addAll(List<T> items);
  T remove(int position);
  T getItem(int position);
  void add(int index, T item);
  int getSize();

  void clear();
}

File: app/src/main/java/io/github/ovso/righttoknow/framework/adapter/BaseAdapterDataModel.kt
package io.github.ovso.righttoknow.framework.adapter

interface BaseAdapterDataModel<T> {
  fun add(item: T)
  fun addAll(items: List<T>)
  fun remove(position: Int): T
  fun getItem(position: Int): T
  fun add(index: Int, item: T)
  val size: Int
  fun clear()
}
